operators/pkg/controller/elasticsearch/driver/default.go

```diff
@@ -545,3 +548,17 @@ func reconcileScriptsConfigMap(c k8s.Client, scheme *runtime.Scheme, es v1alpha1
 
 	return nil
 }
+
+// warnUnsupportedDistro sends an event of type warning if the Elasticsearch Docker image is not a supported
+// distribution by looking at if the prepare fs init container terminated with the UnsupportedDistro exit code.
+func warnUnsupportedDistro(es v1alpha1.Elasticsearch, pods []corev1.Pod, recorder record.EventRecorder) {
```

Comment:
Would a unit test be nice here? Using the reconcileState recorder would help inspecting an event was correctly added in unit tests.

Reply:
I asked myself this question and I couldn't see a relevant unit test.

Here how I see a unit test:
```go
func Test_warnUnsupportedDistro(t *testing.T) {
	recorder := events.NewRecorder()
	require.Equal(t, 0, len(recorder.Events()))
	pods := []corev1.Pod{{Status: corev1.PodStatus{
		InitContainerStatuses: []corev1.ContainerStatus{{
			Name: initcontainer.PrepareFilesystemContainerName,
			LastTerminationState: corev1.ContainerState{
				Terminated: &corev1.ContainerStateTerminated{
					ExitCode: 1,
				}}}}}}}

	warnUnsupportedDistro(pods, recorder)
	require.Equal(t, 0, len(recorder.Events()))

	pods = []corev1.Pod{{Status: corev1.PodStatus{
		InitContainerStatuses: []corev1.ContainerStatus{{
			Name: initcontainer.PrepareFilesystemContainerName,
			LastTerminationState: corev1.ContainerState{
				Terminated: &corev1.ContainerStateTerminated{
					ExitCode: initcontainer.UnsupportedDistroExitCode,
				}}}}}}}

	warnUnsupportedDistro(pods, recorder)
	require.Equal(t, 1, len(recorder.Events()))
}
```

I find the function too trivial to deserve this kind of very basic test.
It makes me think of testing getter and setters...
Do you have something else in your mind?

Reply:
Yeah you're probably right :)